crates/driver/src/infra/api/routes/settle/mod.rs

```diff
@@ -15,15 +15,17 @@ pub(in crate::infra::api) fn settle(router: axum::Router<State>) -> axum::Router
 async fn route(
     state: axum::extract::State<State>,
     _: axum::Json<dto::Solution>,
-) -> Result<axum::Json<dto::Settled>, (hyper::StatusCode, axum::Json<Error>)> {
+) -> Result<(), (hyper::StatusCode, axum::Json<Error>)> {
     let competition = state.competition();
     let auction_id = competition.auction_id().map(|id| id.0);
     let handle_request = async {
         observe::settling();
         let result = competition.settle().await;
         observe::settled(state.solver().name(), &result);
-        let calldata = result?;
-        Ok(axum::Json(dto::Settled::new(calldata)))
+        match result {
+            Err(err) => Err(err.into()),
+            Ok(_) => Ok(()),
+        }
```

Comment:
Unnecessary match statement:
```suggestion
result.map_err(Into::into).map(|_| ())
```